lib/dal-materialized-views/src/luminork/change_set_review.rs

```diff
@@ -0,0 +1,265 @@
+        );
+
+        // Only include components with meaningful changes
+        if corrected_diff_status != ComponentDiffStatus::None {
+            components.push(ComponentReview {
+                component: component_in_list,
+                attribute_diff_trees,
+                action_diffs,
+                corrected_diff_status,
+            });
+        }
+    }
+
+    // Add erased components
+    for (component_id, erased) in erased_components.erased {
+        let action_diffs = action_diffs_by_component
+            .get(&component_id)
+            .cloned()
+            .unwrap_or_default();
+
+        let attribute_diff_trees = to_attribute_diff_trees(&erased.diff.attribute_diffs)?;
+
+        components.push(ComponentReview {
+            component: erased.component,
+            attribute_diff_trees,
+            action_diffs,
+            corrected_diff_status: ComponentDiffStatus::Removed,
+        });
+    }
+
+    // Sort by diff status: Added -> Modified -> Removed
+    components.sort_by_key(|c| match c.corrected_diff_status {
+        ComponentDiffStatus::Added => 0,
+        ComponentDiffStatus::Modified => 1,
+        ComponentDiffStatus::Removed => 2,
+        ComponentDiffStatus::None => 3,
+    });
+
+    Ok(LuminorkChangeSetReview {
+        id: workspace_pk,
+        components,
+    })
+}
+
+/// Groups action diffs by component ID
+fn group_action_diffs_by_component(
+    action_diff_list: &ActionDiffList,
+) -> HashMap<ComponentId, Vec<ActionDiffView>> {
+    let mut result: HashMap<ComponentId, Vec<ActionDiffView>> = HashMap::new();
+
+    for action_diff in action_diff_list.action_diffs.values() {
+        if action_diff.diff_status
+            != si_frontend_mv_types::action::action_diff_list::ActionDiffStatus::None
+        {
+            result
+                .entry(action_diff.component_id)
+                .or_default()
+                .push(action_diff.clone());
+        }
+    }
+
+    result
+}
+
+/// Converts flat attribute diffs to a filtered, flattened list
+fn to_attribute_diff_trees(
+    attribute_diffs: &[(String, AttributeDiff)],
+) -> crate::Result<Vec<AttributeDiffTree>> {
+    let mut result = Vec::new();
+
+    for (path, diff) in attribute_diffs {
+        // Filter out uninteresting diffs
+        if !should_include_diff(path, diff) {
+            continue;
+        }
+
+        result.push(AttributeDiffTree {
+            path: path.clone(),
+            diff: diff.clone(),
+        });
+    }
+
+    // Sort by path for consistent ordering
+    result.sort_by(|a, b| a.path.cmp(&b.path));
+
+    Ok(result)
+}
+
+/// Filters out uninteresting diffs (matching frontend logic)
+fn should_include_diff(path: &str, diff: &AttributeDiff) -> bool {
+    use si_frontend_mv_types::component::component_diff::AttributeSourceAndValue;
+
+    // Filter out internal SI attributes that aren't useful for review
+    if path == "/si/type" || path == "/si/color" {
+        return false;
+    }
```

Comment:
Color is a funny one. Might actually matter to some... I'm okay with this though.